src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -247,12 +229,16 @@ void DeleteQueueIfPresent(string queueName)
         {
             var path = MsmqAddress.Parse(queueName).PathWithoutPrefix;
 
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
+        }
 
-            MessageQueue.Delete(path);
+        string GetReallyLongQueueName()
+        {
+            var maxQueueNameForSetPermissionToWork = 102 - Environment.MachineName.Length;
```

Comment:
After digging into the code some more, I still can't see the usefulness of this test. This test ensures that the setting of the queue permissions does not throw. However, the length it sets in the test is still less than 124 characters, which is a long queue name but still valid, therefore the set permissions call will succeed. 
We have a separate test that ensures that if a queue already exists, then its permission isn't changed. 
What exactly are we testing here? That given a valid but long queue name, set permissions will not throw?